nbconvert/exporters/exporter.py

```diff
@@ -308,4 +308,5 @@ def _preprocess(self, nb, resources):
         #to each preprocessor
         for preprocessor in self._preprocessors:
             nbc, resc = preprocessor(nbc, resc)
+            nbformat.validate(nbc, relax_add_props=True)
```

Comment:
I've posted the same idea on that nbformat issue, but for easy reference, this is what I'd suggest:

```python
try: 
    nbformat.validate(nbc, relax_add_props=True)
except nbformat.ValidationError:
    self.log.error('Notebook is invalid after preprocessor %r', preprocessor)
    raise  # With no args, raise re-raises the same exception
```

It would be neater to have it in the exception message itself, but I think having it in logging is good enough.